### ISSUE TITLE
Couldn't train on my own dataset, because of model diverged with loss = NaN.

### ISSUE THREAD

**Issue by Jiyugaoka** [HiKapok/SSD.TensorFlow#26]
I make my own dateset, and check carefully to ensure all the training image do have at least one bbox. I also set scaffold = None, but it still doesn't work. The ce loss is high, so the total loss is high, and the model diverged with loss = Nan. How can I solve this problem, I have been working with it for some days.  
![2018-07-29](https://user-images.githubusercontent.com/18535374/43362132-d14f2c4e-9314-11e8-942a-686dbaf88a16.png)

**Comment by HiKapok** [comment]
@Jiyugaoka did you try single GPU? set lr to 0 first, if NAN loss still exists, try to set all inputs to ones. eg. features = tf.ones_like(features), if still exists, check wether cls_loss or loc_loss is NAN. BTW ,tf_debug maybe helpful

**Comment by Jiyugaoka** [comment]
sorry for having not reply to your message in time. I have already set lr to 0,and set all inputs to ones. this time the ce loss is smaller than last time as is shown in the picture, this time ce=11.88。But the model is still diverged with loss = NaN. You said tf.debug maybe help,but where to insert a debuger into a estimater, I haven't use tf.debug before, i will be appreciate it if you can give me an example about where to use the debugger to check the ce loss and loc loss.

**Comment by HiKapok** [comment]
@Jiyugaoka see [here](https://www.tensorflow.org/guide/debugger#debugging_tensorflow_estimators). Make sure all your input(includes labels and loc targets) is correct, you can set all of them to constants.

**Comment by HiKapok** [comment]
this issue will be closed, since we have discussed and solved this problem offline vie e-mail

**Comment by Jiyugaoka** [comment]
if you want to know more about how i solved this problem, please refer to my last comment of issue 27#27
